File: docs/code-snippets/first-script-example/count_annotations.py
from cytomine import Cytomine
from cytomine.models import *

# Replace with actual values or -better- use args.
host = 'https://mycytomine.com'
public_key = 'AAA'
private_key = 'ZZZ'
id_term = 0

# Connect to Cytomine
with Cytomine(host, public_key, private_key) as cytomine:
    # Get all the available projects you have access to
    projects = ProjectCollection().fetch()

    # Instantiate a new list
    projects_with_p = ProjectCollection()

    # Filter the projects
    for project in projects:
        if project.name.startswith("P"):
            projects_with_p.append(project)

    # Get the term
    term = Term().fetch(id_term)

    # Count annotations having that term in filtered projects
    count = 0
    for project in projects_with_p:
        annotations = AnnotationCollection(project=project.id, term=term.id).fetch()
        count += len(annotations)

    print("There are {} annotations with term {}".format(count, term.name))

File: docs/code-snippets/first-script-example/CountAnnotations.java
import be.cytomine.client.*;
import be.cytomine.client.collections.*;
import be.cytomine.client.models.*;

public class AnnotationCounter {
  public static void main(String[] args){

    // Replace with actual values or -better- use args.
    String publicKey = "AAA";
    String privateLey = "ZZZ";
    String cytomineCoreUrl = "https://mycytomine.com/"; //Fictive: use your Cytomine server
    Long termId = 0L;

    // Connection to Cytomine
    Cytomine.connection(cytomineCoreUrl, publicKey, privateKey);

    // Get all the available projects you have access to
    Collection<Project> projects = Collection.fetch(Project.class);

    // Instantiate a new list
    List projectsStartingWithP = new ArrayList<>();

    // Filter the projects
    for (int i= 0; i<projects.size();i++){
      Project project = projects.get(i);
      if(project.get("name").startsWith("P") {
        projectsStartingWithP.add(project);
      }
    }

    // Get the term
    Term term = new Term().fetch(termId);

    // Count annotations having that term in filtered projects
    AnnotationCollection ac;
    int count = 0;
    for (Project p : projectsStartingWithP){
      ac = AnnotationCollection.fetchByTermAndProject(term, p);
      count += ac.size();
    }

    System.out.println("There are " + new String(count) + " annotations with term " + term.getStr("name"))
  }
}